webrtc/src/track/track_local/track_local_static_rtp.rs

```diff
@@ -42,6 +44,83 @@ impl TrackLocalStaticRTP {
             .iter()
             .all(|b| b.sender_paused.load(Ordering::SeqCst))
     }
+
+    /// write_rtp_with_extensions writes a RTP Packet to the TrackLocalStaticRTP
+    /// If one PeerConnection fails the packets will still be sent to
+    /// all PeerConnections. The error message will contain the ID of the failed
+    /// PeerConnections so you can remove them
+    ///
+    /// If the RTCRtpSender direction is such that no packets should be sent, any call to this
+    /// function are blocked internally. Care must be taken to not increase the sequence number
+    /// while the sender is paused. While the actual _sending_ is blocked, the receiver will
+    /// miss out when the sequence number "rolls over", which in turn will break SRTP.
+    ///
+    /// Extensions that are already configured on the packet are overwritten by extensions in
+    /// `extensions`.
+    pub async fn write_rtp_with_extensions(
+        &self,
+        p: rtp::packet::Packet,
+        extensions: &[rtp::extension::HeaderExtension],
+    ) -> Result<usize> {
+        let mut n = 0;
+        let mut write_errs = vec![];
+
+        let bindings = {
+            let bindings = self.bindings.lock().await;
+            bindings.clone()
+        };
+        let packet_iterator = RepeatNIterator::new(p, bindings.len());
```

Comment:
Not quite feeling this `RepeatNIterator`. It hides the fact that we're doing `.clone()` for every iteration.

It would be clearer if we just loop over the binding and do an explicit `.clone()` for each loop.

Reply:
The reason I did this was actually to avoid one clone. Previously we were taking a `&rtp::packet::Packet` and then always cloning it(even with a single track binding). With this I've changed it so we take it by value and clone `N - 1` times where `N` is the number of track bindings. Since the common case is to have a single track binding this means no more cloning.

An iterator felt like the most elegant way to achieve this "clone except the last time" behaviour

Reply:
So you want to save one clone, and therefore we need the cognitive load of understanding a bespoke implemented iterator instead of spelling it out in code? Not sure I agree with that. Code is a liability, less code = good.

Reply:
Hmm yeah fair. There's a method that does exactly this in `itertools` but I didn't want to pull in a whole dependency for just that iterator hence building it ourselves. Point taken though, let me rethink